dbms/src/Flash/Pipeline/PipelineBuilder.h

```diff
@@ -0,0 +1,100 @@
+// Copyright 2023 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <Flash/Pipeline/Pipeline.h>
+
+#include <vector>
+
+namespace DB
+{
+class PipelineIdGenerator
+{
+public:
+    UInt32 nextID()
+    {
+        return current_id++;
```

Comment:
```suggestion
        return ++current_id;
```

Reply:
The pipeline id starts at 0, so I think `current_id++` is more appropriate.